stl/inc/numeric

```diff
@@ -469,6 +487,30 @@ _CONSTEXPR20 _OutIt adjacent_difference(const _InIt _First, const _InIt _Last, _
     const auto _ULast = _STD _Get_unwrapped(_Last);
     auto _UDest       = _STD _Get_unwrapped_n(_Dest, _STD _Idl_distance<_InIt>(_UFirst, _ULast));
     if (_UFirst != _ULast) {
+        if constexpr (_Iterators_are_contiguous<_InIt, _OutIt> && !_Iterator_is_volatile<_InIt>
+                      && is_trivially_copyable_v<_Iter_value_t<_InIt>>) {

```

Comment:
Is `!_Iterator_is_volatile<_InIt>` or `is_trivially_copyable_v<_Iter_value_t<_InIt>>` meaningful here, given we're not preforming any trivial bitwise copy in this branch?

Reply:
I'm not sure. 

The Effects as in [[adjacent.difference]/4](https://eel.is/c++draft/adjacent.difference#4) specifically requiring making a copy at certain time, which we do later. So the difference may be observable. But it is not observable, if the memory is not volatile and the copying is trivial.

Reply:
I think these checks make sense. If we're altering how we read the input in any way, we should guard against `volatile` (which is super rare, so it's not like it's a big loss). And if we're altering how we copy/move values, requiring trivially copyable seems proper. I'm not too worried about someone having non-trivially-copyable types that would nevertheless be eligible for auto-vectorization, since this is purely a performance improvement and it's not like we're going to pessimize their existing codegen.